autogpt_platform/backend/schema.prisma

```diff
@@ -19,45 +19,148 @@ model User {
   updatedAt DateTime @updatedAt
 
   // Relations
-  AgentGraphs                  AgentGraph[]
+  AgentGraphs                  Agent[]
   AgentGraphExecutions         AgentGraphExecution[]
   AgentGraphExecutionSchedules AgentGraphExecutionSchedule[]
   AnalyticsDetails             AnalyticsDetails[]
   AnalyticsMetrics             AnalyticsMetrics[]
   UserBlockCredit              UserBlockCredit[]
+  ConfiguredAgents             ConfiguredAgent[]
+  UserAgents                   UserAgents[]
+
+  // User Group relations
+  Groups                 UserGroup[]              @relation("GroupUsers")
+  AdminGroups            UserGroup[]              @relation("GroupAdminUsers")
+  UserAccounting         UserAccounting[]
+  UserSubscription       UserSubscription[]
+  Profile                Profile[]
+  StoreListing           StoreListing[]
+  StoreListingSubmission StoreListingSubmission[]
 
   @@index([id])
   @@index([email])
 }
 
+model UserGroup {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  name         String
+  description  String
+  groupIconUrl String?
+
+  users      User[] @relation("GroupUsers")
+  adminUsers User[] @relation("GroupAdminUsers")
+
+  agents       Agent[]
+  Profile      Profile[]
+  StoreListing StoreListing[]
+
+  @@index([name])
+}
+
 // This model describes the Agent Graph/Flow (Multi Agent System).
-model AgentGraph {
+model Agent {
   id        String    @default(uuid())
   version   Int       @default(1)
   createdAt DateTime  @default(now())
   updatedAt DateTime? @updatedAt
 
   name        String?
   description String?
-  isActive    Boolean @default(true)
-  isTemplate  Boolean @default(false)
 
   // Link to User model
-  userId String
-  user   User   @relation(fields: [userId], references: [id], onDelete: Cascade)
+  createdByUserId String
+  createdByUser   User   @relation(fields: [createdByUserId], references: [id], onDelete: Cascade)
```

Comment:
If a user creates the agent in a group and then they delete their account, the agent will be deleted even if the group persists, correct?

Reply:
And by extension this would delete it from the marketplace

Reply:
The plan is to make it so it is not possible to delete an entry in the database, they are just marked as delete. Then we run our own cleanup logic periodically.

When the agent becomes a group agent, someone in the group with the correct permissions needs to delete it.

The user can delete the agent out of there library, which is stored in the UserAgents table